idea-plugin/src/main/kotlin/arrow/meta/ide/dsl/utils/IdeUtils.kt

```diff
@@ -101,6 +110,33 @@ fun <F : CallableDescriptor> F.returns(ktFunction: KtNamedFunction, types: Kotli
 inline fun <reified K : PsiElement> K.replaceK(to: K): K? =
   replace(to).safeAs()
 
+// fixme use ViewProvider's files instead?
+@Suppress("UNCHECKED_CAST")
+fun <F : PsiFile> List<VirtualFile>.files(project: Project): List<F> =
```

Comment:
Isn't this returning all PsiFiles, regardless of the type of `F`? 
`F` isn't reified and the concrete type of `F` is erased at runtime, afaik.

Reply:
Nop this will only filter files which have the correct Type `F`.